lib/Kubectl/CLIWrapper.pm

```diff
@@ -6,27 +6,36 @@ package Kubectl::CLIWrapper {
 
   our $VERSION = '0.02';
 
-  has server => (is => 'ro', isa => 'Str');
-  has username => (is => 'ro', isa => 'Str');
-  has password => (is => 'ro', isa => 'Str');
-  has token => (is => 'ro', isa => 'Str');
+  has kubectl => (is => 'ro', isa => 'Str', default => 'kubectl');
+  has config => (is => 'ro', isa => 'Str', predicate => 'has_config');
+  has namespace => (is => 'ro', isa => 'Str', predicate => 'has_namespace');
+  has password => (is => 'ro', isa => 'Str', predicate => 'has_password');
+  has server => (is => 'ro', isa => 'Str', predicate => 'has_server');
+  has token => (is => 'ro', isa => 'Str', predicate => 'has_token');
+  has username => (is => 'ro', isa => 'Str', predicate => 'has_username');
+
   has insecure_tls => (is => 'ro', isa => 'Bool', default => 0);
-  has namespace => (is => 'ro', isa => 'Str', default => 'default');
 
-  has kubectl => (is => 'ro', isa => 'Str', default => 'kubectl');
+  has kube_options => (
+        is      => 'ro',
+        isa     => 'ArrayRef[Str]',
+        lazy    => 1,
+        builder => 'build_options',
+  );
 
-  has kube_options => (is => 'ro', isa => 'ArrayRef[Str]', lazy => 1, default => sub {
+  sub build_options {
     my $self = shift;
-
     my %options = ();
-    $options{ server } = $self->server if (defined $self->server);
-    $options{ username } = $self->username if (defined $self->username);
-    $options{ password } = $self->password if (defined $self->password);
-    $options{ namespace } = $self->namespace;
-    $options{ 'insecure-skip-tls-verify' } = 'true' if ($self->insecure_tls);
+
+    $options{server}     = $self->server    if $self->has_server;
+    $options{username}   = $self->username  if $self->has_username;
+    $options{password}   = $self->password  if $self->has_password;
+    $options{namespace}  = $self->namespace if $self->has_namespace;
+    $options{kubeconfig} = $self->config    if $self->has_config;
```

Comment:
I'm not sure of changing the name of the kubeconfig command line option for config. I did that with the insecure_tls option, and I regret it (since it's more intuitive that these options align). Can you please change the name, and document the option together with username, password, etc?

Reply:
Done :+1: